addons/mail/models/mail_mail.py

```diff
@@ -480,6 +484,31 @@ def _send(self, auto_commit=False, raise_exception=False, smtp_session=None):
                     except Exception:
                         pass
 
+                # Compute email size and transform attachment in downloadable link in the body if it exceeds the limit
+                estimated_email_size = (
+                        len(bytes(json.dumps(headers), "utf-8")) * BASE_64_SIZE_OVERHEAD_RATIO +
+                        len(bytes(body, "utf-8")) * BASE_64_SIZE_OVERHEAD_RATIO +
+                        sum(attachment['file_size'] for attachment in
+                            attachments.sudo().read(['file_size'])) * BASE_64_SIZE_OVERHEAD_RATIO +
```

Comment:
why do you need to multiply by a base64 ratio ?

(if really needed I would do this instead: (a + b + c) * ratio)

Reply:
Actually this is the worst case, I think. There are different encoding and as far as I know base64 is the worst of them: 6 bits (64 characters) are encoded in 8 bits which leads to a ratio of 8/6. I have factored it as suggested.